wherehows-web/app/constants/datasets/shared.ts

```diff
@@ -0,0 +1,52 @@
+import Route from '@ember/routing/route';
+import { assert } from '@ember/debug';
+import { set } from '@ember/object';
+import DatasetController from 'wherehows-web/controllers/datasets/dataset';
+
+/**
+ * Defines id strings for page tabs available on the dataset page,
+ * these also match up with their respective route names
+ * @type {string}
+ */
+enum Tabs {
+  Properties = 'properties',
+  Access = 'access',
+  Comments = 'comments',
+  Schema = 'schema',
+  Ownership = 'ownership',
+  Compliance = 'compliance',
+  SampleData = 'sample',
+  Relations = 'relations'
+}
+
+/**
+ * Sets the tab selection property on the provided route with the currently selected tab
+ * @param {Route} route the route instance to update
+ * @param {Tabs} tabSelected identifier for the selected tab
+ * @returns {Tabs}
+ */
+const setTabSelectedOnAncestorController = (route: Route, tabSelected: Tabs): Tabs => {
+  const { routeName, controllerFor } = route;
+  assert('route should be a descendant of datasets.dataset', !routeName.indexOf('datasets.dataset.'));
```

Comment:
Would using `indexOf` here really be appropriate? If the `routeName` does not contain `datasets.dataset` then you will get `-1` which would still technically be a truthy value and pass the assertion, but if it exists at index `0` then you get a falsy value and will fail the assertion.

Reply:
@cptran777 there's a `!` infront of the `routeName.indexOf` operation, it will only be truthy when dataset.datasets is `0`. Hope that's clear?